inference-engine/samples/benchmark_app/utils.hpp

```diff
@@ -8,15 +8,119 @@
 #include <vector>
 #include <map>
 
+namespace benchmark_app {
+    struct InputInfo {
+        InferenceEngine::Precision precision;
+        InferenceEngine::SizeVector shape;
+        std::string layout;
+        bool isImage() const;
+        bool isImageInfo() const;
+        size_t getDimentionByLayout(char character) const;
+        size_t width() const;
+        size_t height() const;
+        size_t channels() const;
+        size_t batch() const;
+        size_t depth() const;
+    };
+    using InputsInfo = std::map<std::string, InputInfo>;
```

Comment:
looks strange that **IE sample** use new structure different from IE ones. As far as I can see, you need only new methods, can such methods be outside of the class and work on IE structures?

Reply:
@ilya-lavrenov Having own duplicate structure is main goal of this MR.  By default, we still working with IE information, but in some cases IE gives wrong data and in this case we ask user set appropriate parameters and we need to keep this provided data somehow. Nothing should be changed in IE when user set input layouts except interpretation of them from application side.